lib/open_project/translations/models/git_repository.rb

```diff
@@ -0,0 +1,73 @@
+require_relative '../helpers/run_command'
+
+class GitRepository
+  include RunCommand
+
+  def initialize(uri, path)
+    @uri = uri
+    @path = path
+  end
+
+  def clone_or_pull
+    if File.directory? File.join(@path, '.git')
+      within_repo do
+        run_command "git fetch"
+      end
+    #elsif File.directory? @path
+      # maybe we should:
+      # raise "Cannot checkout #{@uri} to #{@path}. Directory already exists but has no .git folder."
+      # or ask if the directory should be overwritten
+    else
+      run_command "git clone #{@uri} #{@path}"
+    end
+  end
+
+  def checkout(ref)
+    within_repo do
+      # todo this does not need to be a branch..
+      @branch = ref
```

Comment:
If I see it correctly, you only need the `@branch` instance var for the output in the `merge` method. If that is true, you can use the output of the shellout `git rev-parse --abbrev-ref HEAD` to get the branch name.